python/cornserve/cli/tasklib_explorer.py

```diff
@@ -0,0 +1,127 @@
+"""Utilities to discover tasks and descriptors in cornserve_tasklib.
+
+This module scans the installed `cornserve_tasklib` package to produce
+deployment-ready entries for unit tasks, composite tasks, and task execution
+descriptors. The CLI consumes these entries to create requests to the gateway.
+"""
+
+from __future__ import annotations
+
+from typing import Any, Tuple, get_args, get_origin
+
+import base64
+import importlib
+import inspect
+import pkgutil
+
+
+def _camel_to_kebab(name: str) -> str:
+    out: list[str] = []
+    for i, ch in enumerate(name):
+        if ch.isupper() and i > 0 and (
+            not name[i - 1].isupper() or (i + 1 < len(name) and not name[i + 1].isupper())
+        ):
+            out.append("-")
+        out.append(ch.lower())
+    return "".join(out)
+
+
+def discover_tasklib() -> tuple[
+    list[dict[str, Any]],
+    list[dict[str, Any]],
+    list[dict[str, Any]],
+]:
+    """Discover unit/composite tasks and descriptors from cornserve_tasklib.
+
+    Returns:
+        A tuple of (unit_task_entries, composite_task_entries, descriptor_entries),
+        where each item is a list of dictionaries ready to be serialized into the
+        gateway deployment request payloads.
+    """
+    try:
+        import cornserve_tasklib  # noqa: F401  # ensure package importable
+        from cornserve.task.base import Task, UnitTask
+        from cornserve.task_executors.descriptor.base import TaskExecutionDescriptor
+    except Exception as e:  # pragma: no cover - bubbled to CLI
+        raise ImportError(f"Failed to import cornserve_tasklib or cornserve core modules: {e}") from e
+
+    unit_task_entries: list[dict[str, Any]] = []
+    composite_task_entries: list[dict[str, Any]] = []
+    descriptor_entries: list[dict[str, Any]] = []
+
+    module_source_cache: dict[str, str] = {}
+
+    def get_module_source_b64(module) -> str:
+        if module.__name__ not in module_source_cache:
+            src = inspect.getsource(module)
+            module_source_cache[module.__name__] = base64.b64encode(src.encode("utf-8")).decode("ascii")
```

Comment:
Just wondering, does it have to be base64? Can just UTF-8 string be problematic?